### PR TITLE
release/0.4.0

### PR DESCRIPTION
**Pull request by thescientist13**
## Related Issue
The minimum needed atomic changeset to deliver / release the initial 'Plugins / Packages work.
- #24
- #182 
- #179 
- #183 
- #187 (+ #178 , #140)
- #138 
- #193 
- #207 
- #213 (+ #16 ) 
- #221 
- #9 

As this is a release branch, the following branch maintenance should be followed
- PRs _into_ this branch should be squashed merged
- Rebase this branch when all work is done
- Merge (_NO_ squash) into master when ready

## Summary of Changes
1. [RFC] Integrated Lerna as a monorepo
1. [RFC] Index Hooks Plugins API
1. Google Analytics Plugin + introduce Composite Plugins concept
1. [RFC] Webpack Build Plugins
1. Polyfills Plugin:  #187 (+ pull deps from npm #178) 
1. Fixing what appears to have been some long standing race condition issues in our build, and consistencies partial exacerbated by the introduction of #187 .  Also absorbed #16 as part of the enhancement.
1. [RFC] Web Components loader

So there's more JS now because of Google Analytics, but even supporting IE11 the main bundle size went down about 30KB.  Once we can [decouple the polyfills](https://github.com/ProjectEvergreen/greenwood/issues/193), evergreen only projects will have a super small initial payload size!  I'm expecting to see some pretty good results though from Alpha Testing the Getting Started repo.

Lighthouse score is about the same, but not bad for all around for full browser support down including IE11.  🎉 

#### BEFORE
before-network
before-lighthouse

#### AFTER
after-network
after-lighthouse

## TODO
1. ~~Evaluate final plugins API (see #179) and document~~ made [an issue](https://github.com/ProjectEvergreen/greenwood/issues/209) for tracking 
1. ~~Performance Testing~~
1. ~~Alpha Testing `alpha.0`~~ 
    - https://github.com/thescientist13/greenwood-getting-started/pull/7
    - https://github.com/thegreenhouseio/www.thegreenhouse.io/pull/128
    - #208 ?
1. ~~`alpha.1`~~
1. ~~_[RFC] Polyfill feature detection: #9 * (could easily be done as a post release PR)_~~
1. - `alpha.2`

## Questions / Thoughts
1. ~~I don't think browserslist is implemented.  Should be done as part of #9? 😬~~ - DONE! 
1. ~~Might be good to create a ticket for PWA / Manifest plugin~~ Done #194 
1. ~~Should we deprecate the 404 page?  Maybe make it as a plugin?~~ (defer to #176 )
1. ~~remove `<title>` tag in _index.html_?~~ defer to #128

### CODE REVIEW

File: www/pages/plugins/webpack.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## Webpack Plugins
+
+> This is an expiremental API! [YMMV](http://onlineslangdictionary.com/meaning-definition-of/your-mileage-may-vary)
```
Comment (hutchgrant) [inline www/pages/plugins/webpack.md]: > expiremental

Spelling

File: www/pages/plugins/webpack.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+## Compatibility
+Although we can't promise all webpack plugins will work, we will try and document which ones we have had success with either as part of the CLI or one of our composite plugins.
+| Name  | Notes  |
+|---|---|
+| [BannerPlugin](https://webpack.js.org/plugins/banner-plugin/) | Used as an example in unit tests for `@greenwood/cli`. |
+| [CopyWebpackPlugin](https://github.com/webpack-contrib/copy-webpack-plugin/) | Used as part of our [Polyfills Plugin (`@greenwood/plugin-polyfills`)](https://github.com/ProjectEvergreen/greenwood/tree/master/packages/plugin-polyfills) to copy `@webcomponentsjs/*` polyflls files to Greenwood's output directory. (`publicDir`).
+
+
+## Known Issues
+Currently there are no known issues (yet).  🤞
```
Comment (hutchgrant) [inline www/pages/plugins/webpack.md]: >  🤞

File: www/pages/plugins/index.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+## Plugins
+
+At it's core, Greenwood provides a CLI and some configuration options to enable users to develop and build their projects quickly and simply from markdown.  However, more complex sites and use cases there will come a need to be able to extend the default functionality of Greenwood to support additional capabilities like:
```
Comment (hutchgrant) [inline www/pages/plugins/index.md]: > At it's core, Greenwood provides a CLI and some configuration options to enable users to develop and build their projects quickly and simply from markdown. 

3 ands in that sentence. Could we restructure that while maintaining the same description?

Comment (thescientist13) [reply www/pages/plugins/index.md]: Yeah, good call.  Will take another pass at this one.

**Pull request merged**